File: Haskell/new_approach/new_approach.hs
reverseNum :: String -> String
reverseNum [] = []
reverseNum xs = [last xs] ++ (reverseNum (init xs))

reverseNum' :: String -> String
reverseNum [] = []
reverseNum (x:xs) = reverseNum xs ++ [x]

main = do
	print(reverseNum "123456789098765432123456789")
	print(reverseNum' "987654321234567890987654321")

File: Objective-C/new_approach/new_approach.m
#include <Foundation/Foundation.h>
#include "Reverse.h"

int main(int argc, const char * argv[])
{
    @autoreleasepool
    {
        Reverse * rev = [[Reverse alloc] init];
        NSString * num = @"123456789";
        NSString * res = [rev reverseNum:num];
        NSLog(@"%@", res);
    }
}
